File: color2gray/rgb.py
import cv2

# Fotoğrafın tam yolu
img_path = r"C:\Users\Ebrar Celik\Desktop\goruntu_renklendirme\goruntu_isleme\color2gray\siyah_beyaz.png"

# Siyah-beyaz fotoğrafı oku
gray = cv2.imread(img_path, cv2.IMREAD_GRAYSCALE)

deneme deneme

if gray is None:
    print("⚠️ Fotoğraf bulunamadı! Dosya yolunu kontrol et.")
else:
    print("✅ Fotoğraf yüklendi, boyut:", gray.shape)
    # Colormap uygula
    color_img = cv2.applyColorMap(gray, cv2.COLORMAP_JET)

    # Sonucu göster
    cv2.imshow("Renklendirilmis", color_img)
    cv2.waitKey(0)
    cv2.destroyAllWindows()

File: color2gray/rgb.cpp
#include <opencv2/opencv.hpp>
#include <iostream>
using namespace cv;
using namespace std;

int main() {
    string path = "test.png";
    cout << "Dosya yolu: " << path << endl;

    Mat image = imread(path);
    if (image.empty()) {
        cout << "Resim yüklenemedi! Lütfen dosya yolunu kontrol et." << endl;
        system("dir"); 
        return -1;
    }

    cout << "Orijinal resim boyutu: " << image.cols << "x" << image.rows << endl;

    // ---- Renkli resmi griye dönüştür ----
    Mat grayImage;
    cvtColor(image, grayImage, COLOR_BGR2GRAY);

    // ---- Sonuçları göster ----
    imshow("Orijinal", image);
    imshow("Siyah Beyaz", grayImage);

    waitKey(0);
    return 0;
}
